monty/dev.py

```diff
@@ -42,8 +43,47 @@ def deprecated(
 
     def _convert_date(date: tuple[int, int, int]) -> datetime:
         """Convert date as int tuple (yyyy, MM, dd) to datetime type."""
+
         return datetime(*date)
 
+    def deadline_warning() -> None:
+        """Raise CI warning after removal deadline in code owner's repo."""
+
+        def _is_in_owner_repo() -> bool:
+            """Check if is running in code owner's repo.
+            Only generate reliable check when `git` is installed and remote name
+            is "origin".
+            """
+
+            try:
+                # Get current running repo
+                result = subprocess.run(
+                    ["git", "config", "--get", "remote.origin.url"],
```

Comment:
if you use `git` with https instead of ssh, the URL returned by `git config remote.origin.url` has this format:

```
https://github.com/materialsproject/pymatgen
```

so the `.lstrip` below won't work as expected. 

```py
os.getenv("GITHUB_REPOSITORY") in owner_repo
```

would be better than

```py
owner_repo == os.getenv("GITHUB_REPOSITORY")
```

Reply:
Thanks! I was not aware of this difference. Also there is `GitHub Cli` which seems to depend on the protocol chosen (SSH or HTTPS). What about adding another `lstrip(https://github.com/)`? I feel safer to do a complete match but `in` seems much cleaner (though with some chance of being false position).

Reply:
i don't think there's chance of a false positive. `git remote` URLs don't have query params and where else would an unexpected `org/repo` (e.g. `materialsproject/pymatgen`) slip in?

Reply:
Well as `monty` is designed to be a general purpose package, I'm afraid there might be repo like `org/repo` and `org/repo-docs` or such. In such case, `org/repo` in `org/repo-docs` (or any other repo with a suffix) would be `True`

Reply:
good point! ignore what i said 😅

Reply:
I think I would still need a reliable unit test...which is kind of tricky. I totally have no idea how to simulate different CI environment (in code owner's repo or not ), other than manually alter environment variables. I would think about this later.

Reply:
yes, manually changing env vars is the right way to test this. you can use the `pytest` [`monkeypatch`](https://docs.pytest.org/en/latest/reference/reference.html#pytest.MonkeyPatch.setenv) fixture of `unittest.patch.dict` to modify env vars in a test. they will be auto-reverted when the test finishes

Reply:
Thanks. Would try that soon 😃 .